### PR TITLE
Add pod status check and update help message

### PR DESCRIPTION
**Pull request by ishitasequeira**
#### Description:
- During upgrade, the terminating pod would also be in the list of pods to validate the success of an upgrade and the upgrade would always fail due to the terminated pod.
- Better printing of help message

### CODE REVIEW

**Comment by jgwest** [comment]
E2E test failed on a known issue for which a fix has now been merged.

/retest

**Comment by jgwest** [comment]
```

Summarizing 1 Failure:
  [FAIL] SnapshotEnvironmentBinding Reconciler E2E tests Testing SnapshotEnvironmentBinding Reconciler. [It] Should recreate GitOpsDeployment, if Binding still exists but GitOpsDeployment is deleted.
  /go/src/github.com/redhat-appstudio/managed-gitops/tests-e2e/core/snapshotenvironmentbinding_test.go:370
Ran 89 of 92 Specs in 3493.193 seconds
FAIL! -- 88 Passed | 1 Failed | 0 Pending | 3 Skipped 
```

File: utilities/upgrade-service/install-upgrade.sh [hunk 1]
```diff
@@ -105,12 +118,13 @@ QUAY_USERNAME=redhat-appstudio
...
 GIT_REVISION="main"
 
-while getopts ':i:u:r:' OPTION; do
+while getopts ':i:u:r:h:' OPTION; do
   case "$OPTION" in
     i) IMG=${OPTARG};;
     u) QUAY_USERNAME=${OPTARG};;
     r) GIT_REVISION=${OPTARG};;
-    ?) echo "Available flag options are:\n[-i] to provide gitops service image (default: quay.io/${QUAY_USERNAME}/gitops-service:latest)\n[-u] to provide QUAY registry username (default: redhat-appstudio)"; exit;
+    h) print_help; exit;
```
Comment (anandf) [inline utilities/upgrade-service/install-upgrade.sh:126]: having an exit code is preferred `exit 0` to indicate that ther was no error and `exit 1` for scenarios with invalid option.

**Review by anandf — APPROVED** [review]

**Review by jgwest — APPROVED** [review]
Thanks @ishitasequeira, and thanks @anandf for reviewing!

**Pull request merged**